openedx/stanford/djangoapps/auth_lagunita/api.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+"""
+Integrate with the Marketo API
+"""
+from __future__ import unicode_literals
+import json
+import logging
+
+import requests
+
+
+log = logging.getLogger('auth_lagunita')
+
+
+class MarketoApi(object):
+    """
+    Create a client to access the API
+    """
+
+    def __init__(
+            self,
+            base_url,
+            access_token=None,
+            client_id=None,
+            client_secret=None,
+    ):
+        self.access_token = access_token
+        self.base_url = base_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+
+    def refresh_access_token(self):
+        """
+        Grab a new auth_token from the API
+
+        They expire after some period of time.
+        We could examine the response and determine the session life...
+        Instead, we just request a new one whenever the API tells us ours
+        has expired.
+        """
+        # pylint: disable=line-too-long
+        url = "{base_url}/identity/oauth/token?grant_type=client_credentials&client_id={client_id}&client_secret={client_secret}".format(
+            base_url=self.base_url,
+            client_id=self.client_id,
+            client_secret=self.client_secret,
+        )
+        # pylint: enable=line-too-long
+        log.debug("url: %s", url)
+        access_token = None
+        try:
+            response = requests.get(url)
+        except requests.exceptions.HTTPError as exception:
```

Comment:
Should we catch the base class `requests.exceptions.RequestException` to be more robust? https://stackoverflow.com/a/16511493